### PR TITLE
cde-gtk-theme: add ability to override palette

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

In the past to change colors of this theme you must do
```
 mkdir -p ~/.themes
 cp -Lr --no-preserve=mode /run/current-system/sw/share/themes/cdetheme ~/.themes/
 chmod +x ~/.themes/cdetheme/scripts/switchtheme
 cd ~/.themes/cdetheme/scripts/
 ./switchtheme ../palettes/Crimson.dp 4 0 0 false true false
```

Now its enough to do
```
cde-gtk-theme.override { palette = "Crimson"; numOfColors = 4; }
```

cc @romildo @NickHu @sauyon

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/cdetheme/default.nix [hunk 1]
```diff
@@ -11,16 +15,26 @@ stdenv.mkDerivation rec {
...
+    xvfb_run
+  ];
 
-  installPhase = ''
-    mkdir -p $out/share/themes
-    cp -r cdetheme $out/share/themes
-    patchShebangs $out/share/themes/cdetheme/scripts/switchtheme
-    wrapPythonProgramsIn "$out/share/themes/cdetheme/scripts" "$out $pythonPath"
+  buildPhase = ''
+    export HOME=`pwd`
```
Comment (SuperSandro2000) [inline pkgs/data/themes/cdetheme/default.nix:31]:
```suggestion
    export HOME=$(pwd)
```

File: pkgs/data/themes/cdetheme/default.nix [hunk 2]
```diff
@@ -11,16 +15,26 @@ stdenv.mkDerivation rec {
...
+    cd cdetheme/scripts
+    substituteInPlace switchtheme \
+      --replace "Globals.themesrcdir=Globals.themedir" "Globals.themesrcdir='$HOME/cdetheme'" \
+      --replace "Globals.themedir=os.path.join(userhome,'.themes','cdetheme')" "Globals.themedir='$out/share/themes/cdetheme'" \
```
Comment (SuperSandro2000) [inline pkgs/data/themes/cdetheme/default.nix:35]: I am not sure here but shouldn't $out be evaluated here?

File: pkgs/data/themes/cdetheme/default.nix (hunk 1 above)
Comment (gnidorah) [reply pkgs/data/themes/cdetheme/default.nix]: @SuperSandro2000 Thanks!

File: pkgs/data/themes/cdetheme/default.nix (hunk 2 above)
Comment (gnidorah) [reply pkgs/data/themes/cdetheme/default.nix:32]: @SuperSandro2000 Sorry, don't get. It is evaluated here like expected

**Pull request closed without merging**